pkg/prometheus/rules.go

```diff
@@ -355,11 +354,7 @@ func ValidateRule(promRule monitoringv1.PrometheusRuleSpec) []error {
 		if group.PartialResponseStrategy == "" {
 			continue
 		}
-		if _, ok := thanostypes.PartialResponseStrategy_value[strings.ToUpper(group.PartialResponseStrategy)]; !ok {
-			return []error{
-				fmt.Errorf("invalid partial_response_strategy %s value", group.PartialResponseStrategy),
-			}
-		}
+		// TODO(slashpai): Remove this assignment after v0.65 since this is handled at CRD level
```

Comment:
Sorry for misleading you, I see that ValidateRule() is also called by the admission webhook so it's ok to check that the PartialResponseStrategy value is correct.
But I wonder if we don't have a bug in the current code since it always resets the value to ""?

Reply:
Looks like that since it will always set to `""` after validation, thanos will always use default value only rather than what user set. IIUC the assignment statement was done for Prometheus to ignore that?. Not sure if this feature was used much otherwise I assume there should have been bug reports :)

Reply:
> Not sure if this feature was used much otherwise I assume there should have been bug reports :)

yes probably but i'll check with a cluster to verify our understanding is correct.

Reply:
I will review :)

Reply:
We reviewed this offline